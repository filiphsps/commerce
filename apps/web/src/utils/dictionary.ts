import 'server-only';

import type { Shop } from '@/api/shop';
import type { LanguageCode, Locale, LocaleDictionary } from '@/utils/locale';

const stub = { cart: {}, common: {} };

export const dictionaries: Record<Lowercase<LanguageCode>, () => Promise<LocaleDictionary>> = {
    af: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ak: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    am: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ar: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    as: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    az: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    be: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    bg: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    bm: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    bn: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    bo: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    br: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    bs: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ca: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ce: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ckb: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    cs: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    cu: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    cy: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    da: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    de: () => import('@/i18n/de.json').then((module) => module.default) as any,
    dz: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ee: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    el: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    en: () => import('@/i18n/en.json').then((module) => module.default) as any,
    eo: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    es: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    et: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    eu: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    fa: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ff: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    fi: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    fil: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    fo: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    fr: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    fy: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ga: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    gd: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    gl: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    gu: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    gv: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ha: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    he: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    hi: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    hr: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    hu: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    hy: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ia: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    id: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ig: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ii: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    is: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    it: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ja: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    jv: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ka: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ki: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    kk: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    kl: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    km: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    kn: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ko: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ks: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ku: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    kw: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ky: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    la: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    lb: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    lg: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ln: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    lo: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    lt: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    lu: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    lv: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    mg: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    mi: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    mk: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ml: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    mn: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    mo: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    mr: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ms: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    mt: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    my: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    nb: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    nd: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ne: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    nl: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    nn: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    no: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    om: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    or: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    os: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    pa: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    pl: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ps: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    pt_br: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    pt_pt: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    pt: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    qu: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    rm: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    rn: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ro: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ru: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    rw: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sa: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sc: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sd: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    se: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sg: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sh: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    si: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sk: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sl: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sn: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    so: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sq: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sr: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    su: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    sv: () => import('@/i18n/sv.json').then((module) => module.default) as any,
    sw: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ta: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    te: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    tg: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    th: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ti: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    tk: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    to: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    tr: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    tt: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ug: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    uk: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    ur: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    uz: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    vi: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    vo: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    wo: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    xh: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    yi: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    yo: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    zh_cn: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    zh_tw: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    zh: () => new Promise((resolve) => resolve(stub)) as any /* TODO */,
    zu: () => new Promise((resolve) => resolve(stub)) as any /* TODO */
};

export type DictionaryLanguageCode = keyof typeof dictionaries;

/**
 * Get dictionary for locale.
 * @todo Handle templates.
 *
 * @param {Locale} locale - Locale to get dictionary for.
 * @returns {Promise<LocaleDictionary>} Promise with dictionary.
 */
export const getDictionary = async (props: { shop: Shop; locale: Locale } | Locale): Promise<LocaleDictionary> => {
    let locale: Locale, shop: Shop | undefined;

    if (Object.hasOwn(props, 'shop')) {
        const temp = props as { shop: Shop; locale: Locale };
        locale = temp.locale;
        // eslint-disable-next-line unused-imports/no-unused-vars
        shop = temp.shop;
    } else {
        locale = props as Locale;
    }

    // TODO: Fetch tenant-specific dictionary if it exists and shop is defined.
    return dictionaries[locale?.language?.toLowerCase() as DictionaryLanguageCode]?.() ?? {};
};
