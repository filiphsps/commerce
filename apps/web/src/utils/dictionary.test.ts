import type { DictionaryLanguageCode } from '@/i18n/dictionary';
import { dictionaries } from '@/i18n/dictionary';
import { describe, expect, it } from 'vitest';

describe('utils', () => {
    describe('dictionary', () => {
        it('should have a getter for each language code', () => {
            const languageCodes = Object.keys(dictionaries) as DictionaryLanguageCode[];
            languageCodes.forEach((languageCode) => {
                expect(typeof dictionaries[languageCode]).toBe('function');
            });
        });

        it('should return a promise that resolves to a dictionary for each language code', async () => {
            const languageCodes = Object.keys(dictionaries) as DictionaryLanguageCode[];
            for (const languageCode of languageCodes) {
                const dictionary = await dictionaries[languageCode]();
                expect(typeof dictionary).toBe('object');
                expect(typeof dictionary.common).toBe('object');
                expect(typeof dictionary.cart).toBe('object');
            }
        });

        it('should return a promise that resolves to a dictionary with the correct keys for each language code', async () => {
            const languageCodes = Object.keys(dictionaries) as DictionaryLanguageCode[];
            for (const languageCode of languageCodes) {
                const dictionary = await dictionaries[languageCode]();
                const keys = Object.keys(dictionary);
                expect(keys).toContain('common');
                expect(keys).toContain('cart');
            }
        });
    });
});
