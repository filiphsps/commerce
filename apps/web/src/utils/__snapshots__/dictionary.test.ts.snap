// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 1`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 2`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 3`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 4`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 5`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 6`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 7`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 8`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 9`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 10`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 11`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 12`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 13`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 14`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 15`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 16`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 17`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 18`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 19`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 20`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 21`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 22`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 23`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 24`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 25`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 26`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 27`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 28`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 29`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 30`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 31`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 32`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 33`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 34`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 35`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 36`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 37`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 38`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 39`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 40`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 41`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 42`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 43`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 44`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 45`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 46`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 47`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 48`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 49`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 50`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 51`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 52`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 53`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 54`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 55`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 56`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 57`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 58`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 59`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 60`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 61`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 62`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 63`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 64`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 65`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 66`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 67`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 68`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 69`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 70`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 71`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 72`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 73`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 74`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 75`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 76`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 77`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 78`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 79`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 80`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 81`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 82`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 83`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 84`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 85`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 86`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 87`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 88`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 89`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 90`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 91`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 92`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 93`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 94`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 95`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 96`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 97`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 98`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 99`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 100`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 101`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 102`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 103`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 104`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 105`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 106`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 107`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 108`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 109`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 110`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 111`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 112`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 113`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 114`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 115`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 116`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 117`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 118`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 119`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 120`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 121`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 122`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 123`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 124`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 125`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 126`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 127`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 128`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 129`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 130`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 131`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 132`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 133`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 134`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 135`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 136`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 137`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 138`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 139`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 140`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 141`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 142`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 143`] = `"object"`;

exports[`utils > dictionary > should return a promise that resolves to a dictionary for each language code 144`] = `"object"`;
